src/components/authentication/Login.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react';
+import {
+  Grid,
+  Paper,
+  Avatar,
+  TextField,
+  Button,
+  Typography,
+  Link,
+  InputAdornment,
+} from '@mui/material';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import GoogleIcon from '@mui/icons-material/Google';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+import { useUserContext, signInWithGmail } from '../../context/userContext';
+import IconButton from '@mui/material/IconButton';
+import '../../App.css';
+
+const paperStyle = {
+  padding: 20,
+  height: '60vh',
+  width: '100%',
+  margin: '0 auto',
+};
+const avatarStyle = { backgroundColor: '#1bbd7e' };
+const btnStyle = { margin: '0.5rem 0' };
+
+function Login({ handleChange }) {
+  const [loginEmail, setLoginEmail] = useState('');
+  const [loginPassword, setLoginPassword] = useState('');
+  const [values, setValues] = useState({
+    password: '',
+    showPassword: false,
+  });
+
+  const handleChangePass = (prop) => (event) => {
+    setValues({ ...values, [prop]: event.target.value });
+  };
+
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
```

Comment:
if you want to change the state depend on *previous state*, use arrow function instead, i.e. `prev => !prev`